argus/backend/service/testrun.py

```diff
@@ -0,0 +1,420 @@
+                               run_id, test_id, old_assignee)
+                old_assignee = None
+        self.create_run_event(
+            kind=ArgusEventTypes.AssigneeChanged,
+            body={
+                "message": "Assignee was changed from \"{old_user}\" to \"{new_user}\" by {username}",
+                "old_user": old_assignee_user.username if old_assignee else "None",
+                "new_user": new_assignee_user.username if new_assignee else "None",
+                "username": g.user.username
+            },
+            user_id=g.user.id,
+            run_id=run.id,
+            release_id=test.release_id,
+            group_id=test.group_id,
+            test_id=test.id
+        )
+        return {
+            "test_run_id": run.id,
+            "assignee": str(new_assignee_user.id) if new_assignee_user else None
+        }
+
+    def get_run_comment(self, comment_id: UUID):
+        try:
+            return ArgusTestRunComment.get(id=comment_id)
+        except ArgusTestRunComment.DoesNotExist:
+            return None
+
+    def get_run_comments(self, run_id: UUID):
+        return sorted(ArgusTestRunComment.filter(test_run_id=run_id).all(), key=lambda c: c.posted_at)
+
+    def post_run_comment(self, test_id: UUID, run_id: UUID, message: str, reactions: dict, mentions: list[str]):
+        message_stripped = strip_html_tags(message)
+
+        mentions = set(mentions)
+        for potential_mention in re.findall(self.RE_MENTION, message_stripped):
+            if user := User.exists_by_name(potential_mention.lstrip("@")):
+                mentions.add(user)
+
+        test: ArgusTest = ArgusTest.get(id=test_id)
+        plugin = self.get_plugin(test.plugin_name)
+        release: ArgusRelease = ArgusRelease.get(id=test.release_id)
+        comment = ArgusTestRunComment()
+        comment.message = message_stripped
+        comment.reactions = reactions
+        comment.mentions = [m.id for m in mentions]
+        comment.test_run_id = run_id
+        comment.release_id = release.id
+        comment.user_id = g.user.id
+        comment.posted_at = time.time()
+        comment.save()
+
+        run: PluginModelBase = plugin.model.get(id=run_id)
+        build_number = get_build_number(build_job_url=run.build_job_url)
+        for mention in mentions:
+            params = {
+                "username": g.user.username,
+                "run_id": comment.test_run_id,
+                "test_id": test.id,
+                "build_id": run.build_id,
+                "build_number": build_number,
+            }
+            self.notification_manager.send_notification(
+                receiver=mention.id,
+                sender=comment.user_id,
+                notification_type=ArgusNotificationTypes.Mention,
+                source_type=ArgusNotificationSourceTypes.Comment,
+                source_id=comment.id,
+                content_params=params
```

Comment:
Lost parameter for sending email notification:
`                source_message=comment.message,`

Reply:
Fixed